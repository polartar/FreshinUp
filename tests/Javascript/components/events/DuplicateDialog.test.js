import { createLocalVue, shallowMount, mount } from '@vue/test-utils'
import Component from '~/components/events/DuplicateDialog.vue'

describe('Duplicate Dialog component', () => {
  // Component instance "under test"
  let localVue
  describe('Snapshots', () => {
    test('dialog open', () => {
      localVue = createLocalVue()
      const wrapper = mount(Component, {
        localVue: localVue,
        propsData: {
          duplicateDialog: true,
          duplicating: true
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Methods', () => {
    beforeEach(() => {
      localVue = createLocalVue()
    })

    test('emit actions', () => {
      const wrapper = shallowMount(Component, {
        localVue
      })

      const mockValue = true
      wrapper.vm.changeDuplicateDialogue(mockValue)
      expect(wrapper.emitted()['manage-duplicate-dialog']).toBeTruthy()

      wrapper.vm.onDuplicate()
      expect(wrapper.emitted()['manage-duplicate']).toBeTruthy()
    })
  })
})
