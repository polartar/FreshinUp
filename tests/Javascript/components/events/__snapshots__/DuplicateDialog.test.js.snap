// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Duplicate Dialog component Snapshots dialog open 1`] = `
<div
  class="v-card v-sheet theme--light"
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    class="v-progress-linear"
    role="progressbar"
    style="height: 7px;"
  >
    <div
      class="v-progress-linear__background primary"
      style="height: 7px; opacity: 0.3; width: 100%;"
    />
    <div
      class="v-progress-linear__bar"
    >
      <div
        class="v-progress-linear__bar__indeterminate v-progress-linear__bar__indeterminate--active"
      >
        <div
          class="v-progress-linear__bar__indeterminate long primary"
        />
        <div
          class="v-progress-linear__bar__indeterminate short primary"
        />
      </div>
      <!---->
    </div>
  </div>
   
  <div
    class="v-card__title"
  >
    <div
      class="layout row space-between align-center"
    >
      <div
        class="flex"
      >
        <h3>
          Duplicate Event
        </h3>
      </div>
       
      <button
        class="white--text v-btn v-btn--round v-btn--small theme--light grey"
        type="button"
      >
        <div
          class="v-btn__content"
        >
          <div
            class="flex"
          >
            <i
              aria-hidden="true"
              class="v-icon white--text fa fa fa fa-times theme--light"
              style="font-size: 16px;"
            />
          </div>
           
          <div
            class="flex"
          >
            
          Close
        
          </div>
        </div>
      </button>
    </div>
  </div>
   
  <hr
    class="v-divider theme--light"
  />
   
  <div
    class="v-card__text grey--text"
  >
    <small
      class="font-weight-bold"
    >
      SELECT
    </small>
     
    <p>
      Choose what will be carried over to the duplicate event
    </p>
     
    <div
      class="v-input mt-0 mb-0 p-0 v-input--selection-controls v-input--checkbox v-input--is-label-active v-input--is-dirty theme--light"
    >
      <div
        class="v-input__control"
      >
        <div
          class="v-input__slot"
        >
          <div
            class="v-input--selection-controls__input"
          >
            <input
              aria-checked="true"
              aria-label="Basic Information"
              role="checkbox"
              type="checkbox"
              value=""
            />
            <div
              class="v-input--selection-controls__ripple accent--text"
            />
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light accent--text"
            >
              check_box
            </i>
          </div>
          <label
            aria-hidden="true"
            class="v-label theme--light"
            style="left: 0px; position: relative;"
          >
            Basic Information
          </label>
        </div>
        <div
          class="v-messages theme--light"
        >
          <span
            class="v-messages__wrapper"
            name="message-transition"
            tag="div"
          />
        </div>
      </div>
    </div>
     
    <div
      class="v-input mt-0 mb-0 p-0 v-input--selection-controls v-input--checkbox theme--light"
    >
      <div
        class="v-input__control"
      >
        <div
          class="v-input__slot"
        >
          <div
            class="v-input--selection-controls__input"
          >
            <input
              aria-checked="false"
              aria-label="Venue/lovation (coming soon)"
              role="checkbox"
              type="checkbox"
              value=""
            />
            <div
              class="v-input--selection-controls__ripple"
            />
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              check_box_outline_blank
            </i>
          </div>
          <label
            aria-hidden="true"
            class="v-label theme--light"
            style="left: 0px; position: relative;"
          >
            Venue/lovation (coming soon)
          </label>
        </div>
        <div
          class="v-messages theme--light"
        >
          <span
            class="v-messages__wrapper"
            name="message-transition"
            tag="div"
          />
        </div>
      </div>
    </div>
     
    <div
      class="v-input mt-0 mb-0 p-0 v-input--selection-controls v-input--checkbox v-input--is-label-active v-input--is-dirty theme--light"
    >
      <div
        class="v-input__control"
      >
        <div
          class="v-input__slot"
        >
          <div
            class="v-input--selection-controls__input"
          >
            <input
              aria-checked="true"
              aria-label="Fleet Member (coming soon)"
              role="checkbox"
              type="checkbox"
              value=""
            />
            <div
              class="v-input--selection-controls__ripple accent--text"
            />
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light accent--text"
            >
              check_box
            </i>
          </div>
          <label
            aria-hidden="true"
            class="v-label theme--light"
            style="left: 0px; position: relative;"
          >
            Fleet Member (coming soon)
          </label>
        </div>
        <div
          class="v-messages theme--light"
        >
          <span
            class="v-messages__wrapper"
            name="message-transition"
            tag="div"
          />
        </div>
      </div>
    </div>
     
    <div
      class="v-input mt-0 mb-0 p-0 v-input--selection-controls v-input--checkbox v-input--is-label-active v-input--is-dirty theme--light"
    >
      <div
        class="v-input__control"
      >
        <div
          class="v-input__slot"
        >
          <div
            class="v-input--selection-controls__input"
          >
            <input
              aria-checked="true"
              aria-label="Customer (coming soon)"
              role="checkbox"
              type="checkbox"
              value=""
            />
            <div
              class="v-input--selection-controls__ripple accent--text"
            />
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light accent--text"
            >
              check_box
            </i>
          </div>
          <label
            aria-hidden="true"
            class="v-label theme--light"
            style="left: 0px; position: relative;"
          >
            Customer (coming soon)
          </label>
        </div>
        <div
          class="v-messages theme--light"
        >
          <span
            class="v-messages__wrapper"
            name="message-transition"
            tag="div"
          />
        </div>
      </div>
    </div>
  </div>
   
  <hr
    class="v-divider theme--light"
  />
   
  <div
    class="v-card__actions"
  >
    <div
      class="layout row justify-end"
    >
      <button
        class="v-btn theme--light"
        type="button"
      >
        <div
          class="v-btn__content"
        >
          
        Cancel
      
        </div>
      </button>
       
      <button
        class="v-btn theme--light primary"
        type="button"
      >
        <div
          class="v-btn__content"
        >
          
        Duplicate
      
        </div>
      </button>
    </div>
  </div>
</div>
`;
